### ISSUE TITLE
Parsing string values in -D compiler option

### ISSUE THREAD

**Issue by goshabog** [Ericsson/codechecker#505]
Hi
I have compilation option -DPRODUCT_NAME='"SCTP IETF"' for my project. But when I’m using CodeChecker for code analysis, the following  error is raised:
sctp.c:715:1: error: use of undeclared identifier 'SCTP'
REGVERSION_DEF(SCTP)
version.h:66:36: note: expanded from macro 'REGVERSION_DEF'
                                   PRODUCT_NAME,                        \
                                   ^
<command line>:12:22: note: expanded from here
#define PRODUCT_NAME SCTP
Also I see that the following option is used in command line for clang
 -DPRODUCT_NAME="SCTP
Does CodeChecker support such parameter – string with space? 
I am using 5.6 version of CodeChecker
Best regards, Igor

**Comment by whisperity** [comment]
At first glance, this looks like an escaping issue. Do you use a build environment or a script file, or are you calling the `--build` option and supply the compile command bare?

As a workaround, try making a shell script which runs your compiler and run CodeChecker as:

```bash
CodeChecker check --name foobar --build "./myscript.sh"
```

**Comment by gyorb** [comment]
@goshabog did you manage to analyze your project with the suggestions by @whisperity ?
